tils/jquery-slick.mdx

```diff
@@ -0,0 +1,64 @@
+---
+title: Installing jQuery & Slick
+description: A quick guide on how to setup Slick and jQuery to create carousels/sliders.
+author: Sidharth Subramanian
+date: '2021-08-12'
+category: Slick
+---
+
+## What is Slick?   
+
+Slick is a JS framework that's used to create Carousels/Sliders. 
+While it is really powerful and easy-to-use, the setup in a Drupal/PL environment has some nuances that are easily missed.
+This guide aims to make this setup process a little easier, so that you can be up and running with Slick in no time.
+
+## Setup  
+
+Some of you may have tried loading the CSS & JS files in the libraries.yml file,
+only to find out that it didn't work as you thought it would.  
+I've been there, and that's when I found that the method below works as expected for both Drupal and PL.
+
+### Step 1
+
+Let's say you have a component that uses a slider.  
+Place these lines at the top of that component's twig file, but after the ```{{ attach_library() }}``` line if you have any.  
+
+```html
+<link rel="stylesheet" type="text/css" href="//cdn.jsdelivr.net/npm/slick-carousel@latest/slick/slick.css" />
```

Comment:
We should be using libraries to load Slick in Drupal. Adding `link` and `script` tags in an element isn't recommended. To get it working in PL, you can load the libraries in `_head.twig` and `_foot.twig`.

Reply:
What about a case where we only want to load the library when the component is loaded? Is there a way we can handle those? I remember Vivek telling me to load it there, then we decided against it, since it would load these files, even when the component is not used.